WordPressLoginFlow/src/main/java/org/wordpress/android/login/webauthn/PasskeyCredentialsHandler.kt

```diff
@@ -0,0 +1,83 @@
+import com.google.android.gms.fido.fido2.api.common.PublicKeyCredential
+import com.google.android.gms.fido.fido2.api.common.PublicKeyCredentialDescriptor
+import com.google.android.gms.fido.fido2.api.common.PublicKeyCredentialRequestOptions
+import com.google.android.gms.fido.fido2.api.common.PublicKeyCredentialType.PUBLIC_KEY
+import org.wordpress.android.fluxc.network.rest.wpcom.auth.webauthn.WebauthnChallengeInfo
+import org.wordpress.android.fluxc.network.rest.wpcom.auth.webauthn.WebauthnCredentialResponse
+import org.wordpress.android.fluxc.store.AccountStore.FinishWebauthnChallengePayload
+
+interface OnPasskeyRequestReadyListener {
+    fun onPasskeyRequestReady(intentSenderRequest: IntentSenderRequest)
+}
+
+class PasskeyCredentialsHandler(
+    private val userId: String,
+    private val challengeInfo: WebauthnChallengeInfo
+) {
+    fun createIntentSender(
+        context: Context,
+        listener: OnPasskeyRequestReadyListener
+    ) {
+        val options = PublicKeyCredentialRequestOptions.Builder()
+                .setRpId(challengeInfo.rpId)
+                .setAllowList(challengeInfo.allowCredentials.map(::parseToCredentialDescriptor))
+                .setChallenge(challengeInfo.challenge.decodeBase64())
+                .setTimeoutSeconds(challengeInfo.timeout.toDouble())
+                .build()
+
+        Fido.getFido2ApiClient(context)
+                .getSignPendingIntent(options)
+                .addOnSuccessListener {
+                    val intentSender = IntentSenderRequest
+                            .Builder(it.intentSender)
+                            .build()
+
+                    listener.onPasskeyRequestReady(intentSender)
+                }
+    }
+
+    fun onCredentialsAvailable(keyCredential: PublicKeyCredential): FinishWebauthnChallengePayload {
+        return FinishWebauthnChallengePayload().apply {
+            this.mUserId = userId
+            this.mTwoStepNonce = challengeInfo.twoStepNonce
+            this.mClientData = keyCredential.toJson()
+        }
+    }
+
+    private fun parseToCredentialDescriptor(credential: WebauthnCredentialResponse) =
+            PublicKeyCredentialDescriptor(
+                    PUBLIC_KEY.toString(),
+                    credential.id.decodeBase64(),
+                    credential.transports.asParsedTransports()
+            )
+
+    private fun String.decodeBase64(): ByteArray {
+        return Base64.decode(this, Base64.NO_PADDING or Base64.NO_WRAP or Base64.URL_SAFE)
+    }
+
+    private fun List<String>.asParsedTransports(): List<Transport> {
+        return mapNotNull {
+            when (it) {
+                "usb" -> Transport.USB
+                "nfc" -> Transport.NFC
+                "ble" -> Transport.BLUETOOTH_LOW_ENERGY
+                "internal" -> Transport.INTERNAL
+                "hybrid" -> Transport.HYBRID
+                else -> null
+            }
+        }.takeIf { it.isNotEmpty() } ?: listOf(
```

Comment:
@ThomazFB - This comment is only a suggestion, so feel free to ignore it.
You could potentially use `.filterNotNull().ifEmpty {` which iirc, will return a default list if the original one is empty.

Reply:
Thanks for this suggestion! Using the `ifEmpty` looks way better. Since the mapping is already filtering the `notNull`, I replaced the `takeIf` and evils operator with the `ifEmpty`. Fixed in https://github.com/wordpress-mobile/WordPress-Login-Flow-Android/pull/123/commits/39ea47033fadc740fe6a0fffea4e7f086d3a4d29.

Let me know what you think!